daemon-scheduler/pkg/engine/dispatcher_test.go

```diff
@@ -0,0 +1,351 @@
+// Copyright 2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package engine
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/blox/blox/daemon-scheduler/pkg/clients/css/models"
+	mocks "github.com/blox/blox/daemon-scheduler/pkg/mocks"
+	"github.com/blox/blox/daemon-scheduler/pkg/types"
+	"github.com/golang/mock/gomock"
+	"github.com/pborman/uuid"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+type DispatcherTestSuite struct {
+	suite.Suite
+	environmentSvc   *mocks.MockEnvironment
+	deploymentSvc    *mocks.MockDeployment
+	css              *mocks.MockClusterState
+	ecs              *mocks.MockECS
+	deploymentWorker *mocks.MockDeploymentWorker
+}
+
+func (suite *DispatcherTestSuite) SetupTest() {
+	mockCtrl := gomock.NewController(suite.T())
+	suite.environmentSvc = mocks.NewMockEnvironment(mockCtrl)
+	suite.deploymentSvc = mocks.NewMockDeployment(mockCtrl)
+	suite.css = mocks.NewMockClusterState(mockCtrl)
+	suite.ecs = mocks.NewMockECS(mockCtrl)
+	suite.deploymentWorker = mocks.NewMockDeploymentWorker(mockCtrl)
+}
+
+func TestDispatcherTestSuite(t *testing.T) {
+	suite.Run(t, new(DispatcherTestSuite))
+}
+
+func (suite *DispatcherTestSuite) TestUnknownEvent() {
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
+	defer cancel()
+
+	input := make(chan Event)
+	output := make(chan Event)
+	dispatcher := NewDispatcher(ctx,
+		suite.environmentSvc,
+		suite.deploymentSvc,
+		suite.ecs, suite.css,
+		suite.deploymentWorker,
+		input, output,
+	)
+
+	dispatcher.Start()
+	input <- ErrorEvent{
+		Error: errors.New("Unexpected Error"),
+	}
+}
+
+func (suite *DispatcherTestSuite) TestUpdateInProgressDeploymentEventReturnsError() {
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
+	defer cancel()
+
+	input := make(chan Event)
+	output := make(chan Event)
+	dispatcher := NewDispatcher(ctx,
```

Comment:
yes, I tried using suite, but that meant I need to clear the channels. it made sense to just initialize them in the test method to avoid state leakage.